modules/ROOT/pages/teradata-vantage-engine-architecture-and-concepts.adoc

```diff
@@ -0,0 +1,118 @@
+= Teradata Vantage Engine Architecture and Concepts
+:experimental:
+:page-author: Krutik Pathak
+:page-email: krutik.pathak@teradata.com
+:page-revdate: August 3, 2023
+:description: Teradata Vantage Architecture and Concepts.
+:keywords: data warehouses, teradata vantage engine architecture, teradata, vantage, Access Module Processors (AMP), Parsing Engines (PE), Massively Parallel Processing (MPP), Virtual Disk (Vdisks), BYNET
+:tabs:
+:dir: teradata-vantage-architecture-concepts
+
+== Overview
+
+This article explains the underlying concepts of Teradata Vantage engine architecture. All editions of Vantage, including the Primary Cluster in VantageCloud Lake utilize the same engine.  
+
+Teradata's architecture is designed around the Massively Parallel Processing (MPP), shared-nothing architecture, which enables high-performance data processing and analytics. The MPP architecture distributes the workload into multiple vprocs or virtual processors that are a set of software processes running on a node under Teradata Parallel Database Extensions (PDE) within the multitasking environment of the operating system. vproc where query processing takes place is commonly referred to as an Access Module Processor (AMP). Each AMP is isolated from other AMPs and processes the queries in parallel allowing Teradata to process large volumes of data rapidly. 
+
+The major architectural components of the Teradata Vantage engine include the Parsing Engines (PEs), BYNET, Access Module Processors (AMPs), and Virtual Disks (Vdisks). Vdisks are applicable to the AMPs in enterprise platforms or the Primary Cluster of a VantageCloud Lake environment. 
+
+
+image::{dir}/teradata_architecture_major_components.png[Teradata Vantage Major Architectural Components, width=75%]
+
+== Teradata Vantage Engine Architecture Components
+The Teradata Vantage engine consists of the components below:
+
+=== Parsing Engines (PE)
+When a SQL query is run in Teradata, it first reaches the Parsing Engine. The functions of the Parsing Engine are:
+
+* Manage individual user sessions (up to 120). 
+* Check if the objects used in the SQL query exist.  
+* Check if the user has required privileges against the objects used in the SQL query.  
+* Parse and optimize the SQL queries.  
+* Prepare the execution plan to execute the SQL query and pass it to the corresponding AMPs. 
+* Receives the response from the AMPs and sends it back to the requesting client. 
+
+
+=== BYNET 
+BYNET is a system that enables component communication. The BYNET system provides high-speed bi-directional broadcast, multicast, and point-to-point communication and merge functions. It performs three key functions: coordinating multi-AMP queries, reading data from multiple AMPs, regulating message flow to prevent congestion, and processing platform throughput. These functions of BYNET make Vantage highly scalable and achieve Massively Parallel Processing (MPP) capability.  
```

Comment:
```suggestion
BYNET is a system that enables component communication. The BYNET system provides high-speed bi-directional broadcast, multicast, and point-to-point communication and merge functions. It performs three key functions: coordinating multi-AMP queries, reading data from multiple AMPs, regulating message flow to prevent congestion, and processing platform throughput. These functions of BYNET make Vantage highly scalable and enable Massively Parallel Processing (MPP) capabilities.  
```